CMakeLists.txt

```diff
@@ -698,7 +698,11 @@ if(UNIX)
     # MacOS ld outputs useless warnings like
     # ld: warning: -macosx_version_min not specificed, assuming 10.7
     # suppress them with -w.
-    set(DEFAULT_HOST_LD_FLAGS "-dylib -w -lm")
+    if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
+        set(DEFAULT_HOST_LD_FLAGS "-dynamiclib -w -lm")
+    else()
+        set(DEFAULT_HOST_LD_FLAGS "-dylib -w -lm")
+    endif()
```

Comment:
@franz, `dynamiclib` has to be used depending on whether clang or the linker is used in https://github.com/pocl/pocl/blob/8c36c380afdfeebf9216627fa97a829f7a3d6cd5/lib/CL/devices/common.c#L198
If `clang` is used, `-dynamiclib` has to be given and if `ld` is used `-dylib` has to be used.